### ISSUE TITLE
Projections with findBy select can't / doesn't trim fields

### ISSUE THREAD

**Issue by SledgeHammer01** [spring-projects/spring-data-jpa#3416]
- Spring Boot 3.2.4
- Java 21

I have a `Film` entity with about 6 - 7 properties, one of which looks like this:

```
  @Setter(AccessLevel.NONE)
  @OrderBy("lastName")
  @OneToMany(cascade = CascadeType.REFRESH, orphanRemoval = true, fetch = FetchType.EAGER)
  @Fetch(FetchMode.JOIN)
  @BatchSize(size = 101) // 100 + 1 to account for pagination
  @JoinTable(
      name = "film_actor",
      joinColumns = {@JoinColumn(name = "film_id")},
      inverseJoinColumns = {@JoinColumn(name = "actor_id")})
  private List<Actor> actors = new ArrayList<>();
```

I'm using a JPA repo with the Querydsl predicate executor. `Film` is the entity type. If I do a normal `findAll()`, the first query that gets generated is:

`select f1_0.film_id,f1_0.replacement_cost,f1_0.description,f1_0.rating,f1_0.release_year,f1_0.title from film f1_0`

I'm experimenting with projections, by doing this:

```
public interface FilmDto {
  int getFilmId();
  String getTitle();
}
```

`List<FilmDto> l = this.filmsRepository.findBy(new BooleanBuilder(), q -> q.as(FilmDto.class).all());`

This is fine and "works", but I still get the same select and in fact, its ALSO still doing the second query to pull in the actors even though the projection doesn't want them.

`select f1_0.film_id,f1_0.replacement_cost,f1_0.description,f1_0.rating,f1_0.release_year,f1_0.title from film f1_0`

`q.project()` doesn't seem to have any effect on the selected fields.

If I go a bit more low level and try to do something like:

`query.select(Projections.fields(FilmDto.class, QFilm.film.filmId, QFilm.film.title))`

That works and slims down the select, but no way to include the actors since that's a 1:M.

Basically, the projection functionality as it is in JPA where its pulling down all fields in the entity and dropping them during the projection seems redundant with something like ModelMapper or MapStruct and is even "lesser" imo since with ModelMapper and MapStruct mapping a Film to a FilmDto, you don't need to mess with proxies since class projection is not supported in Spring Data at the moment and you can even do further customization.

As a side note, this IS supported in the Mongo implementation (I get its a document vs. relational), but I can do something like:

`query.select(Projections.fields(FilmDto.class, QFilm.film.filmId, QFilm.film.title, QFilm.film.actors))`

And it works as expected where it adds a projection clause into the Mongo query to just those properties.
